### PR TITLE
Push Message Emitter

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
Helper code that takes an object and works who to send push messages to.

Tested for events, untested for schedules.

Tested on android, not tested on IOS.

push flow on the server

screenshot 2018-06-04 16 16 50

### CODE REVIEW

File: client/src/push/fcm.js [hunk 1]
```diff
@@ -43,6 +43,8 @@ class FCMClient {
 
     self.notificationListener = this.FCN.onNotification(
       (notification) => {
+        console.log(notification);
```
Comment (sdunster) [inline client/src/push/fcm.js]: dont commit debug code (unless this was intentional)

File: server/src/push/emitters.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            sendPush({
+              devices: results,
+              message: `Schedule #${schedule.id} Created\n${schedule.name}`,
+            });
+          });
+        });
+      });
+    },
+    pushEventToGroup(event) {
+      Group.findById(event.groupId).then((group) => {
```
Comment (sdunster) [inline server/src/push/emitters.js]: i'm fairly certain you can traverse these relationships in one query using sequelize. Look up how to a join... You should be able to get a complete list of devices given an event's groupId with only one roundtrip. If it's too much effort don't worry - premature optimisation is a waste of time. Stating intentions with a comment or TODO can help :)

File: client/src/push/fcm.js (hunk 1 above)
Comment (OSPFNeighbour) [reply client/src/push/fcm.js]: intentional. Have no use for the data in this PR but its important to see the full data that FCN got

**Comment by OSPFNeighbour** [comment]
line #10 in emitters is the sendTest push and works just like it did before.

File: server/src/push/emitters.js (hunk 2 above)
Comment (OSPFNeighbour) [reply server/src/push/emitters.js]: Very possible im just worried ill spend 2 days fighting with sequalizer in the process

**Comment by sdunster** [comment]
ok yeah I see, all looks good then - just need to test on iOS.

**Comment by sdunster** [comment]
Sorry I still need to test this - I'll try to get to it tonight while in transit.

**Comment by OSPFNeighbour** [comment]
Any luck? 6 days old

**Comment by sdunster** [comment]
rebase

**Pull request merged**